tests/system/python/api/test_notification.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test notification REST API """
+
+import os
+import subprocess
+import http.client
+import json
+import time
+import urllib
+import pytest
+
+
+__author__ = "Vaibhav Singhal"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+SERVICE = "notification"
+SERVICE_NAME = "Notification Server #1"
+NOTIFY_PLUGIN = "slack"
+NOTIFY_INBUILT_RULES = ["OverMaxRule", "UnderMinRule"]
+DATA = {"name": "Test - 1",
+        "description": "Test4_Notification",
+        "rule": NOTIFY_INBUILT_RULES[1],
+        "channel": NOTIFY_PLUGIN,
+        "enabled": True,
+        "notification_type": "one shot"
+        }
+
+
+class TestNotificationServiceAPI:
+    def test_notification_without_install(self, reset_and_start_foglamp, foglamp_url, wait_time):
+        # Wait for foglamp server to start
+        time.sleep(wait_time)
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/notification')
+        r = conn.getresponse()
+        pytest.xfail("FOGL-2748")
```

Comment:
is this applicable to next assert as well?

Reply:
yes, as per pytest doc,
```...Note that no other code is executed after pytest.xfail call, differently from the marker. That’s because it is implemented internally by raising a known exception.```
We need to call this internally since we want the fixtures (setup conditions) to be called.